benchmark/clBench/clblast/blas2/spr2.cpp

```diff
@@ -0,0 +1,172 @@
+/**************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename spr2.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(char uplo, int size, scalar_t alpha, int incX) {
+  std::ostringstream str{};
+  str << "BM_Spr2<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << uplo << "/" << size << "/" << alpha << "/" << incX;
+  return str.str();
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, ExecutorType* executorPtr, char uplo,
+         int size, scalar_t alpha, int incX, bool* success) {
+  // The counters are double. We convert size to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  const double size_d = static_cast<double>(size * (size + 1) / 2);
+  const double n_d = static_cast<double>(size);
+
+  // Temporarely set incY to 1 for all benchmarks
+  index_t incY = 1;
+
+  state.counters["n"] = n_d;
```

Comment:
Could you remove these counters and use the `blas_benchmark::utils::init_level_2_counters<
      blas_benchmark::utils::Level2Op::spr2, scalar_t>` function here?